### PR TITLE
Cmake fixes

### PR DESCRIPTION
**Pull request by dnakamura**
Small misc cmake fixes. See individual commit messages for details

### CODE REVIEW

**Comment by mgaudet** [comment]
I've no objections :) LGTM 👍

**Comment by rwy0717** [comment]
Lgtm

**Comment by charliegracie** [comment]
No need for a CI build on pLinux and zLinux since they do not use CMake yet.

File: omr/CMakeLists.txt [hunk 1]
```diff
@@ -59,8 +59,6 @@ if(OMR_GC)
 	target_link_libraries(omrcore
 		PRIVATE
 			${OMR_CORE_GLUE_TARGET}
-			omrtrace
```
Comment (charliegracie) [inline omr/CMakeLists.txt]: I thought the OMR core code was responsible for calling the GC startup APIs and also used tracepoints. Doesn't that mean that it needs to link the gc and trace libraries?

Comment (dnakamura) [reply omr/CMakeLists.txt]: It means the end consumer needs to link against those (if they actually end up calling said functions). However the issue is that almost everyone links against omrcore, thus making omrcore depend on anything quickly introduces cycles

**Pull request merged**